network/p2p/block_exchange.go

```diff
@@ -57,7 +57,7 @@ func (e *BlockExchange) GetBlocks(cids ...cid.Cid) network.BlocksPromise {
 }
 
 func (e *BlockExchange) HasBlock(block blocks.Block) error {
-	return e.bs.HasBlock(block)
+	return e.bs.HasBlock(context.Background(), block)
```

Comment:
context is better to be added as an input parameter so that caller may apply a timeout on it.